### PR TITLE
Feature/cu 8ywha8 reload catalog without restarting container

### PR DESCRIPTION
**Pull request by Hemanthkolli**

### CODE REVIEW

**Comment by pubudug** [comment]
I think you would usually wait for an approval before you merge the PR

File: src/main/kotlin/io/meshcloud/dockerosb/DockerOsbApplication.kt [hunk 1]
```diff
@@ -1,11 +1,14 @@
 package io.meshcloud.dockerosb
 
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration
 import org.springframework.boot.autoconfigure.SpringBootApplication
+import org.springframework.boot.context.properties.EnableConfigurationProperties
 import org.springframework.boot.runApplication
+import org.springframework.cloud.servicebroker.autoconfigure.web.Catalog
```
Comment (pubudug) [inline src/main/kotlin/io/meshcloud/dockerosb/DockerOsbApplication.kt:7]: Some unused imports here

File: src/main/kotlin/io/meshcloud/dockerosb/service/GenericCatalogService.kt [hunk 2]
```diff
@@ -36,12 +32,35 @@ class CatalogConfiguration(
...
+  /**
+   * used to provide Catalog object to be used by other services internally
+   */
+  fun getCatalogInternal(): Catalog {
+    getLatestCatalog()
+    return this.catalog
+  }
+
+  override fun getServiceDefinition(serviceId: String?): Mono<ServiceDefinition> {
+    TODO("Not yet implemented")
```
Comment (pubudug) [inline src/main/kotlin/io/meshcloud/dockerosb/service/GenericCatalogService.kt:64]: Is this method not called in any situation?

File: src/test/kotlin/io/meshcloud/dockerosb/ServiceBrokerFixture.kt [hunk 3]
```diff
@@ -1,9 +1,10 @@
 package io.meshcloud.dockerosb
 
-import io.meshcloud.dockerosb.config.CatalogConfiguration
+//import io.meshcloud.dockerosb.config.CatalogConfiguration
```
Comment (pubudug) [inline src/test/kotlin/io/meshcloud/dockerosb/ServiceBrokerFixture.kt:3]: Remove commented out code.

**Review by pubudug — COMMENTED** [review]
If possible try to write a test that will check the dynamic updating of the catalog.

File: src/main/kotlin/io/meshcloud/dockerosb/service/GenericCatalogService.kt (hunk 2 above)
Comment (Hemanthkolli) [reply src/main/kotlin/io/meshcloud/dockerosb/service/GenericCatalogService.kt:64]: As of now its not called in any situation.

**Pull request merged**